rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/ScriptExecutor.java

```diff
@@ -159,20 +161,21 @@ private Node executeScripts() {
             }
         }
 
-        // cleaning the node
-        try {
-            rmnode.clean();
-        } catch (Throwable t) {
-            logger.warn(rmnode.getNodeURL() + " : exception in cleaning", t);
-            logger.warn(rmnode.getNodeURL() + " : pinging the node");
+        if (atLeastOneScriptExecuted) {
             try {
-                // 'pingNode' call can fail with exception if NodeSource was destroyed
-                rmnode.getNodeSource().pingNode(rmnode.getNode());
-            } catch (Throwable pingError) {
-                logger.warn(rmnode.getNodeURL() + " : nodeSource " + rmnode.getNodeSourceName() +
-                            " seems to be removed ", pingError);
+                rmnode.clean();
+            } catch (Throwable t) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)

Reply:
I agree